### PR TITLE
Added additional electra connections

### PR DESCRIPTION
**Pull request by supersteph**

### CODE REVIEW

**Comment by jaeyounkim** [comment]
A different email was used for your commit. Please read [this guide](
https://github.com/tensorflow/models/wiki/Contributor-License-Agreements#troubleshooting-clas) to fix the CLA issue.

**Comment by supersteph** [comment]
@googlebot I signed it!

File: official/nlp/bert/electra_input_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+    """Filter out features to use for pretraining."""
+    x = {
+        'input_mask': record['input_mask'],
+        'input_type_ids': record['segment_ids'],
+        'masked_lm_positions': record['masked_lm_positions'],
+        'masked_lm_ids': record['masked_lm_ids'],
+        'masked_lm_weights': record['masked_lm_weights'],
+    }
+    token, mlm_pos, mlm_l = electra_create_pretraining_data.create_masked_lm_predictions(record['input_word_ids'],
+                                                        mlm_prob, max_predictions_per_seq, vocab,random.Random(FLAGS.random_seed),do_whole_word_mask)
```
Comment (mickeystroller) [inline official/nlp/bert/electra_input_pipeline.py:124]: What is this "mlm_prob" variable and where is it defined?

**Comment by mickeystroller** [comment]
Hi @supersteph, 

Thanks for contributing this code. While I am reviewing the code, I am wondering why you create a new `electra_create_pretraining_data.py`  under the `official/nlp/data/` folder. Is the current `official/nlp/data/create_pretraining_data.py` not enough? It seems you only use the `create_masked_lm_predictions()` function later in the `electra_input_pipeline.py` code, but I don't find any significant change in this function. Do I miss something?

**Comment by supersteph** [comment]
@mickeystroller I was trying to add dynamic masking to the data, but I'll change the code right now to fix those issues. Thanks for pointing them out!

File: official/nlp/modeling/models/electra_pretrainer.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        num_predictions=num_token_predictions,
+        input_width=sequence_output.shape[-1],
+        source_network=network,
+        activation=activation,
+        initializer=initializer,
+        output='logits',
+        name='generator')
+    lm_outputs = tf.identity(self.masked_lm([sequence_output, masked_lm_positions]), "mlm_logits")
+    fake_data, labels = _get_fake_data(inputs[0], lm_outputs, masked_lm_positions)
+    other_output, other_cls_output = discriminator([fake_data, inputs[1], inputs[2]])
```
Comment (mickeystroller) [inline official/nlp/modeling/models/electra_pretrainer.py:108]: Please all more comments in this part. Thanks.

File: official/nlp/modeling/models/electra_pretrainer.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+  def from_config(cls, config, custom_objects=None):
+    return cls(**config)
+
+
+def _get_fake_data(og, predictions, maskedlm_ids):
+  tokids = tf.stop_gradient(tf.math.argmax(predictions,axis=-1,output_type=tf.dtypes.int32))
+  updatedids, mask = _scatter_update(og,maskedlm_ids,tokids)
+  labels = mask*(1-tf.cast(tf.equal(og,updatedids),tf.int32))
+  return updatedids, labels
+def _scatter_update(og, maskedlm_ids, tokids):
```
Comment (mickeystroller) [inline official/nlp/modeling/models/electra_pretrainer.py:135]: What is the "og" variable used in this function and where is it defined?

Comment (supersteph) [reply official/nlp/modeling/models/electra_pretrainer.py:135]: the "og" variable is the original sentence's word ids

**Comment by saberkun** [comment]
@supersteph Thanks!
@mickeystroller has made a version that probably better tested training the electra-base E2E.
We are consolidating that.
